File: inst/Matlab/fdaM/@fd/putcoef.m
function newfdobj = putcoef(fdobj, coef)
%  PUTCOEF   Replaces the coefficient array from FDOBJ.

%  last modified 1 July 1998

  if isa_fd(fdobj)
    newfdobj.coef     = coef;
    newfdobj.basisobj = fdobj.basisobj;
    newfdobj.fdnames  = fdobj.fdnames;
    newfdobj = class(newfdobj, 'fd');
  else
    error('Argument FDOBJ is not a functional data object.');
  end

File: R/putcoef.R
putcoef <- function(coef, fd) {
	#  replace the coefficient matrix in a functional data object
	  
	#  Last modified 10 Feb 2003
  
  	if (inherits(fd, "fd")) {
		fd$coefs <- as.array(coef)
  	}  else {
      	stop("Argument FD is not a functional data object.")
  	}
  	return(fd)
}
